### ISSUE TITLE
[Feature] Implement complex numbers functionalities

### ISSUE THREAD

**Issue by edgarriba** [kornia/kornia#507]
I open this ticket to discuss the design of a module for complex images -> `kornia.geoemtry.linalg.complex`

The idea is to cover basic functionalities for real/imaginari parts. I list in below the potential functions to cover:
- `tensor_to_complex`
- `multiply_complex`
- `sun_complex`
- `subs_complex`
- `div_complex`
- `module_complex`

This is prototyped here:

https://colab.research.google.com/drive/1K__v_vsuMa0eknTLLTZsXR-VhyJGB7IO

**Comment by fschiffers** [comment]
Just wanted to bump this since the newer versions of PyTorch will have significantly improved complex number support. Would be great if Kornia could handle these data types as well!

**Comment by edgarriba** [comment]
@fschiffers feel free to send a PR. You can also join our slack to discuss the api.

**Comment by SravanChittupalli** [comment]
Hey! @edgarriba @shijianjian I would like to implement complex number functionalities. Please assign this to me.

**Comment by edgarriba** [comment]
@SravanChittupalli we have to do a bit of discussion here because pytorch added support for that already after I added this ticket. Do you have a special need / motivation on this end ?

**Comment by fschiffers** [comment]
@edgarriba : It would be great if Kornia could directly handle these complex datatypes. I understand that this might be complicated for some specific functions, but there's definitely a bunch of functions that could be easily implemented for complex support and I think several scientific communities could greatly benefit from this.

**Comment by edgarriba** [comment]
I agree that is simple as I show in the colab but we shouldn’t overlap functionality respect pytorch mainstream 
See: https://pytorch.org/docs/stable/complex_numbers.html
